pkg/config/app_config.go

```diff
@@ -245,16 +233,50 @@ func (c *AppConfig) GetTempDir() string {
 }
 
 func configFilePath(filename string) (string, error) {
-	folder, err := findOrCreateConfigDir()
-	if err != nil {
-		return "", err
+	exists, path := findConfigFile(filename)
+
+	if exists {
+		return path, nil
+	} else {
+		return path, os.MkdirAll(filepath.Dir(path), 0o755)
+	}
+}
+
+func findConfigFile(filename string) (exists bool, path string) {
+	if envConfigDir := os.Getenv("CONFIG_DIR"); envConfigDir != "" {
+		return true, filepath.Join(envConfigDir, filename)
+	}
+
+	legacyConfigPath, err := xdg.SearchConfigFile(filepath.Join("jesseduffield", "lazygit", filename))
+	if err == nil {
+		return true, legacyConfigPath
 	}
 
-	return filepath.Join(folder, filename), nil
+	configFilepath, err := xdg.SearchConfigFile(filepath.Join("lazygit", filename))
+	if err == nil {
+		return true, configFilepath
+	}
+
+	return false, filepath.Join(xdg.ConfigHome, "lazygit")
```

Comment:
When starting lazygit with the `-d` option and without an existing config folder, you get the error `Unable to log to log file: open /Users/stk/Library/Application Support/lazygit: is a directory`.

Looks like this can be fixed by appending `filename` to the path as well:

```suggestion
	return false, filepath.Join(xdg.ConfigHome, "lazygit", filename)
```